docs: Use `inject` for DI (#59835)
angular/angular: adev/src/content/guide/components/inheritance.md

### Forwarding injected dependencies

> If a base class relies on constructor dependency injection, the child class must explicitly pass these

Reviewer: I would rephrase this slightly
```suggestion
If a base class injects dependencies as constructor parameters, the child class must explicitly class these dependencies to `super`.
```
